doc/source/cross-language.rst

```diff
@@ -0,0 +1,246 @@
+  
+  # Call the Java remote function.
+  obj_ref3 = add_function.remote(1, 2)
+  assert ray.get(obj_ref3) == 3
+
+  ray.shutdown()
+  
+Java calling Python
+-------------------
+
+You can write a Python module as follows:
+
+.. code-block:: python
+
+  class Counter(object):
+    def __init__(self):
+        self.value = 0
+
+    def increment(self):
+        self.value += 1
+        return self.value
+
+
+  def add(a, b):
+      return a + b
+
+Then you can create Python actor from Java, or call Python remote function
+from Java:
+
+.. code-block:: java
+
+  import io.ray.api.Ray;
+  import io.ray.api.function.PyActorClass;
+  import io.ray.api.function.PyActorMethod;
+  import io.ray.api.function.PyFunction;
+  import org.testng.Assert;
+
+  public class MyRayApp {
+
+    public static void main(String[] args) {
+      Ray.init();
+      
+      // Define a Python class.
+      PyActorClass actorClass = PyActorClass.of(
+          "<your python module>", "Counter");
+      
+      // Create a Python actor and call actor method.
+      PyActorHandle actor = Ray.actor(actorClass).remote();
+      ObjectRef<Integer> objRef1 = actor.task(
+          PyActorMethod.of("increment", Integer.class)).remote();
+      Assert.assertEquals(objRef1.get(), 1);
+      ObjectRef<Integer> objRef2 = actor.task(
+          PyActorMethod.of("increment", Integer.class)).remote();
+      Assert.assertEquals(objRef2.get(), 2);
+
+      // Define a Python remote function.
+      PyFunction<Integer> addFunction = PyFunction.of(
+          "<your python module>", "add", Integer.class);
+
+      // Call the Python remote function.
+      ObjectRef<Integer> objRef3 = Ray.task(add_function, 1, 2).remote();
+      Assert.assertEquals(objRef3.get(), 3);
+
+      Ray.shutdown();
+    }
+  }
+
+Cross-language data serialization
+---------------------------------
+
+The arguments and return values of ray call can be serialized & deserialized
+automatically if their types are the following:
+  
+  - Primitive data types
+      ===========   =======  =======
+      MessagePack   Python   Java
+      ===========   =======  =======
+      nil           None     null
+      bool          bool     Boolean
+      int           int      Short / Integer / Long / BigInteger
+      float         float    Float / Double
+      str           str      String
+      bin           bytes    byte[]
+      ===========   =======  =======
+
+  - Basic container types
+      ===========   =======  =======
+      MessagePack   Python   Java
+      ===========   =======  =======
+      array         list     Array
+      ===========   =======  =======
+
+  - Ray builtin types
+      - ActorHandle
+
+.. note::
+
+  * Be aware of float / double precision between Python and Java. If Java use a
+    float type to receive the input argument, the double precision Python data
+    will be reduced to float precision in Java.
+
+You can write a Python function which returns the input data:
```

Comment:
Add "The following example shows how to pass these types as parameters and how to return return these types."

Reply:
Thanks.